src/sammich/plugins/reminder.py

```diff
@@ -0,0 +1,113 @@
+import datetime
+import logging
+import re
+import time
+
+import requests
+from dotenv import dotenv_values
+from machine.plugins.base import MachineBasePlugin
+from machine.plugins.decorators import command
+
+secrets = dotenv_values(".secrets")
+
+SLACK_BOT_TOKEN = secrets["SLACK_BOT_TOKEN"]
+
+reminder_pattern = re.compile(r"\[(.*?)\]\s+\[(.+?)\]\s+\[(.+?)\]")
+
+
+class ReminderPlugin(MachineBasePlugin):
+    def parse_command(self, command_text):
+        match = reminder_pattern.match(command_text)
+        if match:
+            channel_name, message, when = match.groups()
+            return channel_name, message, when
+        return None, None, None
+
+    def get_channel_id(self, channel_name):
+        # Use Slack API to get the channel ID from the channel name
+        headers = {
+            "Content-Type": "application/json",
+            "Authorization": f"Bearer {SLACK_BOT_TOKEN}",
+        }
+        response = requests.get("https://slack.com/api/conversations.list", headers=headers)
+        channels = response.json().get("channels", [])
+        for channel in channels:
+            if channel["name"] == channel_name.lstrip("#"):
+                return channel["id"]
+        return None
+
+    def convert_to_timestamp(self, when):
+        try:
+            future_time = datetime.datetime.strptime(when, "%d/%m/%Y %I:%M %p")
+            future_timestamp = int(future_time.timestamp())
+
+            return future_timestamp
+        except ValueError:
+            return None
+
+    def schedule_message(self, channel_id, text, post_at):
+        headers = {
+            "Content-Type": "application/json",
+            "Authorization": f"Bearer {SLACK_BOT_TOKEN}",
+        }
+        data = {
+            "channel": channel_id,
+            "text": text,
+            "post_at": post_at,
+        }
+        response = requests.post(
```

Comment:
I suggest leveraging the abstractions we have from slack-machine instead of `requests` to interact with the Slack API where possible.

Reply:
Sure!